client/src/app/site/base/base-view-model.ts

```diff
@@ -45,6 +45,10 @@ export abstract class BaseViewModel<M extends BaseModel = any>
         return `${this.collectionString}:${this.id}`;
     }
 
+    public get parent(): BaseViewModel<M> {
```

Comment:
Please remove this. Not every view model does have a parent. Also calling this function (if it is not overwritten) creates an endless recursion.

If you need to have some BaseModels with the property parent, create an interface for this.